Cabal/src/Distribution/PackageDescription/Check/Common.hs

```diff
@@ -1,3 +1,5 @@
+{-# LANGUAGE ViewPatterns #-}
```

Comment:
Where is this needed? How verbose would this PR become without it?

Reply:
> View patterns permit calling the view function inside the pattern and matching against the result
> [GHC users guide - View Patterns](https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/view_patterns.html#view-patterns).

It allows this transformation from `Version` to `[Int]` by calling `versionNumbers` and matching `vs` to the `[Int]` (the result of that function application), and other similar transformations.

```Haskell
trailingZero :: Version -> Bool
trailingZero (versionNumbers -> vs)
  | [0] <- vs = False
  | 0 : _ <- reverse vs = True
  | otherwise = False
```

 I get more bang-for-the-buck from the recursive predicates, from `hasLEQUpperBound`, from `hasGTLowerBound` and from `hasTrailingZeroUpperBound`.